hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -370,28 +372,19 @@ public String getLockOwnerInfo(Data key) {
 
     @Override
     public Record loadRecordOrNull(Data key, boolean backup, Address callerAddress) {
-        Object value = mapDataStore.load(key);
-        if (value == null) {
+        Object oldValue = mapDataStore.load(key);
+        if (oldValue == null) {
             return null;
         }
 
-        long ttl = UNSET;
-        if (mapDataStore.isWithExpirationTime()) {
-            MetadataAwareValue loaderEntry = (MetadataAwareValue) value;
-            long proposedTtl = expirationTimeToTtl(loaderEntry.getExpirationTime());
-            if (proposedTtl <= 0) {
-                return null;
-            }
-            value = loaderEntry.getValue();
-            ttl = proposedTtl;
-        }
-        Record record = createRecord(key, value, ttl, UNSET, getNow());
-        markRecordStoreExpirable(ttl, UNSET);
+        long lastLoadedTtl = mapDataStore.getLastLoadedTtl();
```

Comment:
I understand that now, because of concurrency, we need create per-record store wrappers. Although it does move some code out of the record store, I'm thinking is it worth it. I can imagine lots of wrappers are created with high partition counts and it's just to keep the last loaded TTL value and some additional code.

Although it complicates the record store, can we try keeping the code in the record store instead and avoid the map store wrapper?

Reply:
makes sense, i removed wrapper.